### ISSUE TITLE
Node sync error block at #5545118 v.9.3.0

### ISSUE THREAD

**Issue by gitloverr** [paritytech/polkadot#3289]
hi everyone,

I'm getting those errors on my node:

`
```
Thread 'txpool-verifier0' panicked at 'Failed to allocate memory: "Requested allocation size is too large"', /usr/local/cargo/git/checkouts/substrate-7e08433d4c370a21/85fa0ab/primitives/io/src/lib.rs:1068
This is a bug. Please report it at:
        https://github.com/paritytech/polkadot/issues/new
2021-06-17 17:11:51  ⚙️  Syncing  0.0 bps, target=#5547863 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 10.2kiB/s ⬆ 5.1kiB/s
2021-06-17 17:11:56  ⚙️  Syncing  0.0 bps, target=#5547864 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.2kiB/s ⬆ 0.3kiB/s
2021-06-17 17:12:01  ⚙️  Syncing  0.0 bps, target=#5547865 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 4.3kiB/s ⬆ 2.0kiB/s
2021-06-17 17:12:06  ⚙️  Syncing  0.0 bps, target=#5547866 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.8kiB/s ⬆ 0.7kiB/s
[root@polkadot ~]# docker logs polka93 --tail 10
This is a bug. Please report it at:
        https://github.com/paritytech/polkadot/issues/new
2021-06-17 17:11:51  ⚙️  Syncing  0.0 bps, target=#5547863 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 10.2kiB/s ⬆ 5.1kiB/s
2021-06-17 17:11:56  ⚙️  Syncing  0.0 bps, target=#5547864 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.2kiB/s ⬆ 0.3kiB/s
2021-06-17 17:12:01  ⚙️  Syncing  0.0 bps, target=#5547865 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 4.3kiB/s ⬆ 2.0kiB/s
2021-06-17 17:12:06  ⚙️  Syncing  0.0 bps, target=#5547866 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.8kiB/s ⬆ 0.7kiB/s
2021-06-17 17:12:11  ⚙️  Syncing  0.0 bps, target=#5547866 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 65.7kiB/s ⬆ 0.3kiB/s
[root@polkadot ~]#
```
`

the node stopped at 5545118.

what happened?

what should I do?

thank you

**Comment by gitloverr** [comment]
hi everyone,

I'm getting those errors on my node:

`
```
Thread 'txpool-verifier0' panicked at 'Failed to allocate memory: "Requested allocation size is too large"', /usr/local/cargo/git/checkouts/substrate-7e08433d4c370a21/85fa0ab/primitives/io/src/lib.rs:1068
This is a bug. Please report it at:
        https://github.com/paritytech/polkadot/issues/new
2021-06-17 17:11:51  ⚙️  Syncing  0.0 bps, target=#5547863 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 10.2kiB/s ⬆ 5.1kiB/s
2021-06-17 17:11:56  ⚙️  Syncing  0.0 bps, target=#5547864 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.2kiB/s ⬆ 0.3kiB/s
2021-06-17 17:12:01  ⚙️  Syncing  0.0 bps, target=#5547865 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 4.3kiB/s ⬆ 2.0kiB/s
2021-06-17 17:12:06  ⚙️  Syncing  0.0 bps, target=#5547866 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.8kiB/s ⬆ 0.7kiB/s
[root@polkadot ~]# docker logs polka93 --tail 10
This is a bug. Please report it at:
        https://github.com/paritytech/polkadot/issues/new
2021-06-17 17:11:51  ⚙️  Syncing  0.0 bps, target=#5547863 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 10.2kiB/s ⬆ 5.1kiB/s
2021-06-17 17:11:56  ⚙️  Syncing  0.0 bps, target=#5547864 (33 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.2kiB/s ⬆ 0.3kiB/s
2021-06-17 17:12:01  ⚙️  Syncing  0.0 bps, target=#5547865 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 4.3kiB/s ⬆ 2.0kiB/s
2021-06-17 17:12:06  ⚙️  Syncing  0.0 bps, target=#5547866 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 2.8kiB/s ⬆ 0.7kiB/s
2021-06-17 17:12:11  ⚙️  Syncing  0.0 bps, target=#5547866 (34 peers), best: #5545120 (0x02ea…94a2), finalized #5545118 (0x3a85…b719), ⬇ 65.7kiB/s ⬆ 0.3kiB/s
[root@polkadot ~]#
```
`

the node stopped at 5545118.

what happened?

what should I do?

thank you

**Comment by NukeManDan** [comment]
To solve this, please downgrade to Polkadot 0.8.30 and run with the following flags added or changed to the correct option:
```
polkadot --execution native --wasm-execution Compiled`
```

The easiest path is to download (and verify) the `polkadot` binary and run it:
- https://github.com/paritytech/polkadot/releases/tag/v0.8.30

If you choose to build from source, you must [use the rust 1.51.0 release](https://doc.rust-lang.org/edition-guide/rust-2018/rustup-for-managing-rust-versions.html#managing-versions) of the stable and nightly toolchain: 
```bash
# Run in the top level of the Polkadot source working directory
git checkout v0.8.30
rustup override set 1.51.0
cargo build --release
```

## NOTE: There will be a 0.9.5 release very soon

You will be encouraged to use when available. Please join [Polkadot Validator Lounge](https://matrix.to/#/#polkadotvalidatorlounge:web3.foundation)  and monitor for the latest information and questions. 🙏🏼

**Comment by gitloverr** [comment]
hi @NukeManDan  the Polkadot 0.8.30 had the following errors when I tried it in the past (both binary and docker image and 8.30 sync  didn't :  at #3105 )

**Comment by NukeManDan** [comment]
Try again now? With the binary.

**Comment by gitloverr** [comment]
hi @NukeManDan polkadot 0.8.30 had those errors. Node stopped at block 4877027  :

`
```
Thread 'tokio-runtime-worker' panicked at 'Storage root must match that calculated.', /usr/local/cargo/git/checkouts/substrate-7e08433d4c370a21/3655f9b/frame/executive/src/lib.rs:455

This is a bug. Please report it at:

        https://github.com/paritytech/polkadot/issues/new

2021-06-18 23:19:30  Block prepare storage changes error:
RuntimeApiError(Application(Execution(RuntimePanicked("Storage root must match that calculated."))))
2021-06-18 23:19:30  💔 Error importing block 0xf7db8dba85a792ac0ccb54d0aa225e487298f6c4d061f9ecea888d89d3f46cfb: Err(Other(ClientImport("Import failed: Error at calling runtime api: Execution failed: RuntimePanicked(\"Storage root must match that calculated.\")")))
2021-06-18 23:19:31  ⚙️  Syncing  0.0 bps, target=#5564044 (3 peers), best: #4877027 (0x6db3…1d74), finalized #4876800 (0x94ba…9dbb), ⬇ 19.5kiB/s ⬆ 1.7kiB/s
2021-06-18 23:19:36  ⚙️  Syncing  0.0 bps, target=#5564044 (3 peers), best: #4877027 (0x6db3…1d74), finalized #4876800 (0x94ba…9dbb), ⬇ 13.4kiB/s ⬆ 1.3kiB/s
2021-06-18 23:19:41  ⚙️  Syncing  0.0 bps, target=#5564044 (3 peers), best: #4877027 (0x6db3…1d74), finalized #4876800 (0x94ba…9dbb), ⬇ 8.5kiB/s ⬆ 2.4kiB/s
2021-06-18 23:19:46  ⚙️  Syncing  0.0 bps, target=#5564044 (3 peers), best: #4877027 (0x6db3…1d74), finalized #4876800 (0x94ba…9dbb), ⬇ 9.3kiB/s ⬆ 2.9kiB/s
2021-06-18 23:19:51  ⚙️  Syncing  0.0 bps, target=#5564044 (3 peers), best: #4877027 (0x6db3…1d74), finalized #4876800 (0x94ba…9dbb), ⬇ 8.7kiB/s ⬆ 1.3kiB/s

```
`
what should I do?

**Comment by NukeManDan** [comment]
are you running with the prescribed flags as mentioned above?

**Comment by gitloverr** [comment]
hi @NukeManDan 

yes I added the flags and also I sync from scratch...

I start the node like this: 

`./polkadot --name "NODEDOT830" --chain polkadot --rpc-external --ws-external --execution native --wasm-execution Compiled --rpc-cors all `

**Comment by NukeManDan** [comment]
All reports suggest this should be working. What is your machine type and specification?

**Comment by gitloverr** [comment]
hi, @NukeManDan  

but it doesn't work... doesn't sync...

`
Linux - CentOS 8.3.2011

8 GB RAM - ~ 500 GB free storage

`

**Comment by NukeManDan** [comment]
Your specs seem to be sufficient (8 GB might be a bottleneck for your first error, likely not though) - I can try and setup a VM with CentOS to reproduce if needed, but we don't see this on Ubuntu 20.04 LTS for sure (although #3286 might)

Strange I don't see your node "NODEDOT830" on https://telemetry.polkadot.io/#list/Polkadot
Are you reporting to this endpoint (should be default)?
```
polkadot --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0'
```

That should help us check what your node's status is.

**Comment by gitloverr** [comment]
hi @NukeManDan ,

you probably don't see it because I didn't use the telemetry flag and stopped the node ... will switching to ubuntu machine solve the problem? do I understand right? also what is the estimated release date for 0.9.4?

thank you

**Comment by NukeManDan** [comment]
0.9.5 is released, and 0.9.6 is _almost_ released (this is a minor patch) 
I would give 0.9.5 a try

**Comment by gitloverr** [comment]
hi @NukeManDan,

0.9.5 is seems stable. there is no problem, now and node synced.

thank you
